pints/plot.py

```diff
@@ -8,6 +8,201 @@
+    parameters held at the given set of parameters.
+    
+    Arguments:
+
+    `log_likelihood`
+        A class (or that extends from) :class:`pints.LogLikelihood` of the 
+        problem of interest.
+    `parameters`
+        A set of parameters, which the `log_likelihood` will be evaluated at, 
+        with the same dimension as the `log_likelihood` will accept.
+    `boundaries`
+        A class (or that extends from) :class:`pints.Boundaries` which defines 
+        the boundaries of each parameters.  
+    `n_eva`
+        (Optional) The number of evaluation of each parameter.
+    `chain`
+        (Optional) A markov chain of shape `(samples, dimension)`, where 
+        `samples` is the number of samples in the chain and `dimension` is the 
+        number of parameters. This will be used to define the boundaries. Only 
+        provide either `chain` or `boundaries`, but not both.
+
+    Returns a `matplotlib` figure object and axes handle.
+    """
+    import pints
+    import numpy as np
+    try:
+        import matplotlib.pyplot as plt
+    except:
+        raise ImportError("The method pints.plot.trace requires matplotlib")
+
+    n_param = len(parameters)
+    # Check dimension
+    if n_param != log_likelihood.dimension():
+        raise ValueError('Input parameters dimension must match the dimension'
+                        ' of the log_likelihood accept')
+    # Check input
+    if not isinstance(boundaries, pints.Boundaries) and boundaries is not None:
+        raise TypeError('The argument boundaries must be a type of or extended'
+                ' from pints.Boundaries')
+    if chain is None and boundaries is not None:
+        def_bound = False
+    elif chain is not None and boundaries is None:
+        def_bound = True
+    else:
+        raise TypeError('One of the input chain or boundaries (and only one)'
+                        ' is expected')
+
+    fig, axes = plt.subplots(n_param, 1, figsize=(6, 2*n_param))
+    for i, p in enumerate(parameters):
+        axes[i].set_xlabel('Parameter ' + str(1 + i))
+
+        # Generate some x-values near the given parameters
+        if def_bound:
+            mu = np.mean(chain[:,i])
+            sigma = np.std(chain[:,i])
+            xmin = mu - 3*sigma
+            xmax = mu + 3*sigma
+        else:
+            xmin = boundaries.lower()[i]
+            xmax = boundaries.upper()[i]
+
+        x = np.linspace(xmin, xmax, n_eva)
+
+        # Calculate log-likelihood with other parameters fixed
+        y = [log_likelihood(list(parameters[:i]) + [j]
+            + list(parameters[1+i:])) for j in x]
+
+        # Plot
+        axes[i].plot(x, y, c='green', label='Log-likelihood')
+        axes[i].axvline(p, c='blue', label='Input value')
+        axes[i].legend()
+
+    # Add vertical y-label to middle plot
+    #fig.text(0.04, 0.5, 'log Likelihood', va='center', rotation='vertical')
+    axes[int(i/2)].set_ylabel('log Likelihood')
+
+    plt.tight_layout()
```

Comment:
Is it always ok to use this? I get errors / ugly plots some times. Not sure why!